torch/_dynamo/variables/tensor.py

```diff
@@ -659,6 +659,12 @@ def redistribute_fn_with_prim_types(x):
                 **options,
             )
 
+    def rename(self, tx, name):
+        new_name = tx.output.new_var(name)
+        self.proxy.node.name = new_name
```

Comment:
This is not right, `new_var` is based off of `co_varnames` but here the name must not conflict with any preexisting var name in FX graph

Reply:
ah yea, lets make newvar take this into account, should be ez

Reply:
This seems worse. The two allocation mechanisms are unrelated right? Just make another function for what you need. BTW fx has fresh name allocation logic internally thouh I don't remember exactly if it is accessible

Reply:
yeah agreed